models/reviews.model.js

```diff
@@ -95,22 +95,22 @@ exports.fetchAllReviews = (
   }
   const tableQuery = db.query(queryStr, queryVals);
   const allCategories = db.query(`SELECT * FROM categories`);
-  return Promise.all([tableQuery, allCategories]).then(
-    ([resultTable, resultCategories]) => {
-      const endpointQuery = resultTable.rows;
-      const categoryCheck = resultCategories.rows.map((obj) => obj.slug);
-      if (endpointQuery.length) {
-        return endpointQuery.map(review=>{
-          review.comment_count=Number(review.comment_count)
-          return review;
-        });
-      } else if (!endpointQuery.length && !categoryCheck.includes(category)) {
+
+  return allCategories
+    .then(({ rows }) => {
+      if (
+        !rows.map((obj) => obj.slug).includes(category) &&
+        category !== undefined
+      ) {
         return Promise.reject({ status: 404, msg: 'Category not found' });
-      } else {
-        return [];
       }
-    }
-  );
+    })
+    .then(() => {
+      return tableQuery
+        .then(({ rows }) => {
+          return rows;
+        })
```

Comment:
This is the right order but now we are nesting our .thens, try to keep it all in one promise chain at the same level